### PR TITLE
Add support for configuration

### LINKED ISSUE #9: Add configurability
Story: As a server operator, I want to be able to configure ports, db location, and some other items.  Please make this configurable.

Exit Criteria: We are using Viper and Cobra to setup keyserver configurability.

Impact: Winning

### PR DESCRIPTION
**Pull request by schancel**
Summary:
Currently, we have no way of specifying some basic configuration to the server.
This means we can't specify bind address, list of peers, or an HMAC secret. So,
in order to address that we add the use of the viper and cobra libraries for handling
command line arguments, and config files as yaml.

Closes #9

Test Plan:
Run by hand

### CODE REVIEW

File: cmd/keyserverd/main.go [hunk 1]
```diff
@@ -1,34 +1,100 @@
...
+	}
+
+	// Find user information
+	usr, err := user.Current()
 	if err != nil {
 		log.Error().Msg(err.Error())
 		os.Exit(1)
 	}
+
+	rootCmd.Flags().StringP("config", "c", "", "Configuration file")
```
[the remark this answers is not in the document]
Comment (hlb8122) [reply cmd/keyserverd/main.go]: Additionally, toggle for mainnet vs testnet?

File: cmd/keyserverd/main.go [hunk 2]
```diff
@@ -1,34 +1,100 @@
...
+	rootCmd.Flags().StringP("bind", "b", "0.0.0.0:8080", "Bind Address for keyserverd")
+	rootCmd.Flags().StringArrayP("peer", "p", []string{}, "URL to a keyserver peer")
```
[the remark this answers is not in the document]
Comment (schancel) [reply cmd/keyserverd/main.go]: The flag is cumulative.  `--peers foo --peers bar` doesn't make sense.

File: cmd/keyserverd/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (schancel) [reply cmd/keyserverd/main.go]: Why would the metadata for a key change depending on the network?

[the remark this answers is not in the document]
Comment (schancel) [reply cmd/keyserverd/main.go]: Token is tied to a particular key right now, and it doesn't depend on the client logic to swap this out later.

[the remark this answers is not in the document]
Comment (hlb8122) [reply cmd/keyserverd/main.go]: The metadata wouldn't change but
https://github.com/cashweb/keyserver/blob/c900c839e291c8fb3e4c2537c0cff9e6f055b7e1/pkg/keydb/keydb.go#L66
takes mainnet parameters. Suppose you wanted to test by accepting payments on testnet?

[the remark this answers is not in the document]
Comment (schancel) [reply cmd/keyserverd/main.go]: That is the default given no prefix.

**Pull request merged**